### PR TITLE
Remove deprecated trace.StartSpanWithXXX convenience methods

### PR DESCRIPTION
**Pull request by ramonza**

### CODE REVIEW

File: plugin/ocgrpc/grpc_test.go [hunk 1]
```diff
@@ -72,12 +72,12 @@ func TestNewClientStatsHandler(t *testing.T) {
 }
 
 func TestNewServerStatsHandler(t *testing.T) {
-	ctx := context.Background()
-
-	handler := NewServerStatsHandler()
-
-	te := &traceExporter{}
-	trace.RegisterExporter(te)
+	var (
```
Comment (rakyll) [inline plugin/ocgrpc/grpc_test.go]: Can you revert these changes?

In Go, this style is discouraged. var grouping is mostly discouraged unless the variables need to be strictly grouped together for readability. var is entirely discouraged unless it is for the zero value.

File: plugin/ocgrpc/trace_common.go [hunk 2]
```diff
@@ -55,16 +55,23 @@ func (c *clientTraceHandler) TagRPC(ctx context.Context, rti *stats.RPCTagInfo)
 // it finds one, uses that SpanContext as the parent context of the new span.
 //
 // It returns ctx, with the new trace span added.
-func (s *serverTraceHandler) TagRPC(ctx context.Context, rti *stats.RPCTagInfo) context.Context {
-	md, _ := metadata.FromIncomingContext(ctx)
-	name := "Recv" + strings.Replace(rti.FullMethodName, "/", ".", -1)
+func (sh *serverTraceHandler) TagRPC(ctx context.Context, rti *stats.RPCTagInfo) context.Context {
+	var (
```
Comment (rakyll) [inline plugin/ocgrpc/trace_common.go]: ditto

File: examples/grpc/helloworld_server/main.go [hunk 3]
```diff
@@ -59,7 +59,7 @@ func main() {
 
 	// Set up a new server with the OpenCensus
 	// stats handler to enable stats and tracing.
-	s := grpc.NewServer(grpc.StatsHandler(ocgrpc.NewServerStatsHandler()))
```
Comment (ramonza) [inline examples/grpc/helloworld_server/main.go]: This should be in the other PR

**Comment by ramonza** [comment]
@rakyll PTAL

File: plugin/ocgrpc/trace_common.go [hunk 4]
```diff
@@ -58,13 +58,18 @@ func (c *clientTraceHandler) TagRPC(ctx context.Context, rti *stats.RPCTagInfo)
 func (s *serverTraceHandler) TagRPC(ctx context.Context, rti *stats.RPCTagInfo) context.Context {
 	md, _ := metadata.FromIncomingContext(ctx)
 	name := "Recv" + strings.Replace(rti.FullMethodName, "/", ".", -1)
+	var span *trace.Span
 	if s := md[traceContextKey]; len(s) > 0 {
 		if parent, ok := propagation.FromBinary([]byte(s[0])); ok {
-			ctx, _ = trace.StartSpanWithRemoteParent(ctx, name, parent, trace.StartOptions{})
-			return ctx
+			span = trace.NewSpanWithRemoteParent(name, parent, trace.StartOptions{})
```
Comment (rakyll) [inline plugin/ocgrpc/trace_common.go]: I'd move `ctx = trace.WithSpan(ctx, span)` after this line and return.

```
if s := md[traceContextKey]; len(s) > 0 {
	if parent, ok := propagation.FromBinary([]byte(s[0])); ok {
		span = trace.NewSpanWithRemoteParent(name, parent, trace.StartOptions{})
		return trace.WithSpan(ctx, span)
	}
}

ctx, _ = trace.StartSpan(ctx, name) 
return
```
